sdk/servicebus/azure-servicebus/azure/servicebus/aio/_base_handler_async.py

```diff
@@ -112,7 +112,42 @@ def __init__(
     @classmethod
     def _convert_connection_string_to_kwargs(cls, conn_str, **kwargs):
         # pylint:disable=protected-access
-        return BaseHandlerSync._convert_connection_string_to_kwargs(conn_str, **kwargs)
+        host, policy, key, entity_in_conn_str, token, token_expiry = _parse_conn_str(
+            conn_str
+        )
+        queue_name = kwargs.get("queue_name")
+        topic_name = kwargs.get("topic_name")
+        if not (queue_name or topic_name or entity_in_conn_str):
+            raise ValueError(
+                "Entity name is missing. Please specify `queue_name` or `topic_name`"
+                " or use a connection string including the entity information."
+            )
+
+        if queue_name and topic_name:
+            raise ValueError(
+                "`queue_name` and `topic_name` can not be specified simultaneously."
+            )
+
+        entity_in_kwargs = queue_name or topic_name
+        if (
+            entity_in_conn_str
+            and entity_in_kwargs
+            and (entity_in_conn_str != entity_in_kwargs)
+        ):
+            raise ValueError(
+                "The queue or topic name provided: {} which does not match the EntityPath in"
+                " the connection string passed to the ServiceBusClient constructor: {}.".format(
+                    entity_in_conn_str, entity_in_kwargs
+                )
+            )
+
+        kwargs["fully_qualified_namespace"] = host
+        kwargs["entity_name"] = entity_in_conn_str or entity_in_kwargs
+        # This has to be defined seperately to support sync vs async credentials.
+        kwargs["credential"] = cls._create_credential_from_connection_string_parameters(
+            token, token_expiry, policy, key
+        )
+        return kwargs
```

Comment:
I'm wondering if there's a better approach to avoid duplication.

like we could make `_convert_connection_string_to_kwargs` a helper method inside the `_base_handler,py` module, and then we pass the corresponding class type to call the `_create_credential_from_connection_string_parameters` method.